src/test/java/MockTest.java

```diff
@@ -0,0 +1,435 @@
+    }
+
+    @Test
+    public void shouldOperatorSearchPlatePageNotFindPlateWithGivenNr() throws Exception {
+        final String EXAMPLE_PLATE_NR = "12345678";
+
+        mockMvc.perform(post("/operatorPlateSearch").param("plateNr", EXAMPLE_PLATE_NR)
+                .sessionAttr("plate", new Plate()))
+                .andExpect(status().isOk())
+                .andExpect(content().string(containsString(env.getProperty("operatorPlateNotFound.plateNotFound"))));
+    }
+
+    @Test
+    public void shouldDriverStopMeter() throws Exception {
+        final String EXAMPLE_PLATE_NR = "12345678";
+        Plate plate = new Plate();
+        plate.setPlateNr(EXAMPLE_PLATE_NR);
+        plate.setStart(Instant.now());
+        plateRepository.save(plate);
+
+        mockMvc.perform(post("/stopAndPay").param("plateNr", EXAMPLE_PLATE_NR)
+                .param("start", Instant.now().toString())
+                .sessionAttr("plate", plate))
+                .andExpect(status().isOk());
+        assertNotNull(plateRepository.findAll().get(0).getPlateNr());
+    }
+
+    @Test
+    public void shouldDriverNotFindPlateAfterStopMeter() throws Exception {
+        final String EXAMPLE_PLATE_NR = "12345678";
+        Plate plate = new Plate();
+        plate.setPlateNr(EXAMPLE_PLATE_NR);
+        plate.setStart(Instant.now());
+        plate.setEnd(Instant.now());
+        plateRepository.save(plate);
+
+        mockMvc.perform(post("/savePlate").param("plateNr", EXAMPLE_PLATE_NR)
+                .sessionAttr("plate", plate))
+                .andExpect(status().isOk())
+                .andExpect(content().string(containsString(env.getProperty("plateNotFound.plateNotFound"))));
+    }
+
+    private void findOpenPlateWhenAClosedOneIsInRepo(String url, String property) throws Exception {
+        final String EXAMPLE_PLATE_NR = "12345678";
+        Plate plate = new Plate();
+        plate.setPlateNr(EXAMPLE_PLATE_NR);
+        plate.setStart(Instant.now());
+        plate.setEnd(Instant.now());
+        plateRepository.save(plate);
+        plate.setEnd(null);
+        plateRepository.save(plate);
+
+        mockMvc.perform(post(url).param("plateNr", EXAMPLE_PLATE_NR)
+                .param("region", Region.PLN.toString())
+                .sessionAttr("plate", plate))
+                .andExpect(status().isOk())
+                .andExpect(content().string(containsString(env.getProperty(property))));
+    }
+
+    @Test
+    public void shouldDriverFindNewPlateAfterMeterStopThatIsRunning() throws Exception {
+        findOpenPlateWhenAClosedOneIsInRepo("/savePlate", "plateFound.plateFound");
+    }
+
+    @Test
+    public void shouldOperatorFindNewPlateAfterMeterStopThatIsRunning() throws Exception {
+        findOpenPlateWhenAClosedOneIsInRepo("/operatorPlateSearch", "operatorPlateFound.plateFound");
+    }
+
+    private void priceCheck(Instant start, Instant end, boolean isVip) throws Exception {
```

Comment:
Co sądzisz o metodach, które posiadają argument typu boolean?

Reply:
Co ta metoda sprawdza?

Reply:
1. Sądzę, że w takich metodach autor nie chce zobaczyć nulla : P.
Praktyki są różne. Jedni mówią, że dobrą praktyką jest wstawianie Boolean, zważywszy na trzecią "wartość" == null.
Druga praktyka mówi, żeby stosować boolean wtedy kiedy nie chcemy przyjmować nulla, bo program szybszy i mniejsze marnowanie zasobów.
Ja uznałem, że nie ma miejsca na nulla.
Wzorowałem się też dyskusją z:
https://stackoverflow.com/questions/2716002/why-should-i-pass-boolean-as-a-parameter-instead-of-boolean

2. Można by poprawić nazwę tej metody, np. na whenEnteredMeterVipArgs_shouldReturnHTMLWithProperPrice(args).
Jej celem jest sprawdzenie, czy po wywołaniu post z odpowiednio zainicjalizowanym parametrem plate kontroler wywoła adekwatne metody i dostanę prawidłową odpowiedź w postaci html z właściwą wyceną.